### PR TITLE
simulator.h: doxygen style documentation

### PR DESCRIPTION
**Pull request by gemarcano**
- Added doxygen style documentation to the main simulator header.

There are some FIXMEs that I need some input on. I'll highlight them once I create the PR.

This also serves as an example of the kinds of small PRs to expect as I continue to document the rest of the header files. Going forward, would you like single file PRs, or for me to bundle them in some way (say, by folder)?

### CODE REVIEW

File: simulator/core/simulator.h [hunk 1]
```diff
@@ -58,9 +108,41 @@ extern int usetestflash;
...
+ * @param[in] target Target cycle to seek to.
+ *
+ * @returns True on success, false if it was not able to.
+ * @post On success, the simulator was set to the specified target cycle. On
+ * failure, most likely the pipeline and simulator are no longer in sync.
+ */
 bool simulator_state_seek(int target);
 #endif
+
+// FIXME misnomer
```
Comment (ppannuto) [inline simulator/core/simulator.h:129]: Wow, yeah, I had to re-read that code a few times to re-grok what this is doing. Should probably be something closer to `sim_update_and_pause_elapsed_time` -- wordy, but clearer..

File: simulator/core/simulator.h [hunk 2]
```diff
@@ -58,9 +108,41 @@ extern int usetestflash;
...
+ * stop!).
+ *
+ * The value recorded is used to calculate the effective frequency of the
+ * simulator. This function is not meant to be called when the simulator is
+ * already "sleeping".
+ */
 void sim_sleep(void);
+
+//FIXME misnomer
+/** Reset the recorded start time of the simulator.
```
Comment (ppannuto) [inline simulator/core/simulator.h:141]: Again a bit of a poorly named variable, it's not restarting the start time per se, it's really marking the time when the simulator started executing again, which is slightly different I think.

**Review by ppannuto — COMMENTED** [review]
As you can probably see, exception handling was some of the last stuff implemented, and is a bit bolted on in places. The TS port is possibly a good opportunity to think about this more; there are probably some better ways to handle exceptions than what is currently here.

**Comment by ppannuto** [comment]
Re single file vs. folder -- no strong preference. I'll be faster to look over smaller units, but grouping logical changes together is very reasonable.

**Comment by gemarcano** [comment]
Let me know if you'd like for me to review any of the changes (namely, the function renames are likely what might need to change).

**Review by ppannuto — APPROVED** [review]

**Pull request merged**